tests/test_init_and_backend.py

```diff
@@ -0,0 +1,131 @@
+        self.assertEqual(mkosi.strip_suffixes(Path("home.xz/test.xz")), Path("home.xz/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home.xz/test")), Path("home.xz/test"))
+        self.assertEqual(mkosi.strip_suffixes(Path("home.xz/test.txt")), Path("home.xz/test.txt"))
+
+    def test_copy_file(self):
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            dir_path = Path(tmp_dir)
+            file_1 = Path(dir_path) / "file_1.txt"
+            file_2 = Path(dir_path) / "file_2.txt"
+
+            file_1.touch()
+            file_2.touch()
+
+            # Copying two empty files.
+            mkosi.copy_file(file_1, file_2)
+            self.assertTrue(filecmp.cmp(file_1, file_2))
+
+            # Copying content from one file.
+            file_1.write_text("Testing copying content from this file to file_2.")
+            mkosi.copy_file(file_1, file_2)
+            self.assertTrue(filecmp.cmp(file_1, file_2))
+
+            # Giving a non existing path/file.
+            with self.assertRaises(error):
+                mkosi.copy_file("nullFilePath", file_1)
+
+            # Copying when there's already content in both files.
+            file_2.write_text("Testing copying content from file_1 to file_2, with previous data.")
+            mkosi.copy_file(file_1, file_2)
+            self.assertTrue(filecmp.cmp(file_1, file_2))
+
+    def test_first_partition_offset(self):
+        table = backend.PartitionTable()
+        table.grain = 4096
+
+        # Grain = 4096, first_lba = None.
+        self.assertEqual(table.first_partition_offset(), 20480)
+
+        # Grian = 4096, first_lba not None.
+        table.first_lba = 64
+        self.assertEqual(table.first_partition_offset(), 32768)
+        table.first_lba = 0
+        self.assertEqual(table.first_partition_offset(), 0)
+
+        # Grain = 1024 ** 2, first_lba = None.
+        table.grain = 1024 ** 2
+        table.first_lba = None
+        self.assertEqual(table.first_partition_offset(), 1048576)
+
+        # Grain = 1024 ** 2, first_lba not None.
+        table.first_lba = 128
+        self.assertEqual(table.first_partition_offset(), 65536)
+        table.first_lba = 256
+        self.assertEqual(table.first_partition_offset(), 131072)
+
+    def test_last_partition_offset(self):
+        table = backend.PartitionTable()
+        table.grain = 4096
+
+        table.last_partition_sector = 32
+        self.assertEqual(table.last_partition_offset(), 16384)
+        table.last_partition_sector = 16
+        self.assertEqual(table.last_partition_offset(), 8192)
+        table.last_partition_sector = 1
+        self.assertEqual(table.last_partition_offset(), 4096)
+
+        table.last_partition_sector = 0
+        table.first_lba = 64
+        self.assertEqual(table.last_partition_offset(), 32768)
+        table.first_lba = 0
+        self.assertEqual(table.last_partition_offset(), 0)
+
+    def test_footer_size(self):
```

Comment:
Since the `first_partition_offset` can depend on `footer_size` I'd move this test before the one for the former.